### PR TITLE
feat: add possibility to deploy emqx-exporter with FlowFuse helm chart

### PR DESCRIPTION
**Pull request by ppawlowski**
## Description

This pull request adds a possibility to deploy emqx-exporter with FlowFuse Helm chart.
Aditionally, it simplifies the logic behind checking the availability of APIs required by EMQX cluster.

## Related Issue(s)

https://github.com/FlowFuse/CloudProject/issues/559

## Checklist

 - [x] I have read the [contribution guidelines](https://github.com/FlowFuse/flowfuse/blob/main/CONTRIBUTING.md)
 - [ ] Suitable unit/system level tests have been added and they pass 
 - [ ] Documentation has been updated
    - [ ] Upgrade instructions
    - [ ] Configuration details
    - [ ] Concepts
 - [ ] Changes `flowforge.yml`?
    - [ ] Issue/PR raised on `FlowFuse/helm` to update ConfigMap Template
    - [ ] Issue/PR raised on `FlowFuse/CloudProject` to update values for Staging/Production

## Labels

 - [ ] Includes a DB migration? -> add the `area:migration` label

### CODE REVIEW

File: helm/flowforge/templates/_helpers.tpl [hunk 1]
```diff
@@ -109,7 +117,7 @@ Note: The value for key .Values.postgresql.auth.existingSecret is inherited from
 
 {{- define "forge.createSecret" -}}
 {{- if not (and .Values.postgresql.auth.existingSecret 
-    (not (and .Values.forge.email (not .Values.forge.email.smtp.existingSecret)))) -}}
+    (not (and .Values.forge.email ((and .Values.forge.email.smtp (not .Values.forge.email.smtp.existingSecret)))))) -}}
```
Comment (hardillb) [inline helm/flowforge/templates/_helpers.tpl:120]: This should already be merged...

File: helm/flowforge/templates/_helpers.tpl [hunk 2]
```diff
@@ -172,3 +180,16 @@ Configure broker domain
...
+*/}}
+{{- define "emqx.bootstrapApiKeySecret" -}}
+{{- $secretName := "emqx-config-secrets" }}
+{{- $existingSecret := (lookup "v1" "Secret" .Release.Namespace $secretName) | default dict }}
+{{- if and $existingSecret.data (hasKey $existingSecret.data "api_key_secret") }}
+{{- printf $existingSecret.data.api_key_secret | b64dec }}
+{{- else }}
+{{- randAlphaNum 32 -}}
+{{- end }}
+{{- end -}}
```
Comment (hardillb) [inline helm/flowforge/templates/_helpers.tpl:195]: Need to think about this, as we will be needing to use this secret in the forge app soon.

This should be fine as we can do the same trick with the helper container to copy it into the flowforge.yml file

Comment (hardillb) [reply helm/flowforge/templates/_helpers.tpl:195]: Check here, as you are only setting the key value, not the name, but use the `flowfuse` name later

File: helm/flowforge/templates/emqx.yaml [hunk 3]
```diff
@@ -220,8 +222,8 @@ metadata:
 type: Opaque
 data:
     EMQX_DASHBOARD__DEFAULT_PASSWORD: {{ "topSecret" | b64enc | quote }}
-    api-key-name: {{ "flowfuse" | b64enc | quote }} 
-    apit-key-secret:  {{ "verySecret" | b64enc | quote }} 
+    api_key_name: {{ "flowfuse" | b64enc | quote }} 
+    api_key_secret: {{ include "emqx.bootstrapApiKeySecret" . | b64enc | quote }}
```
Comment (hardillb) [inline helm/flowforge/templates/emqx.yaml:226]: OK, you are setting the name here

**Review by hardillb — APPROVED** [review]
Looks OK, just some comments on the secret_key_name, but I think they are OK

File: helm/flowforge/templates/_helpers.tpl (hunk 1 above)
Comment (hardillb) [reply helm/flowforge/templates/_helpers.tpl:120]: yeah, this is the merge conflict

Comment (ppawlowski) [reply helm/flowforge/templates/_helpers.tpl]: Fixed.

**Comment by ppawlowski** [comment]
CI failure is cause by the chart testing tool to reuse values file from older chart version instead of new one (https://github.com/helm/chart-testing/issues/525).

**Pull request merged**